src/ADAL.PCL/Internal-Telemetry/AggregatedDispatcher.cs

```diff
@@ -0,0 +1,81 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.IdentityModel.Clients.ActiveDirectory
+{
+    class AggregatedDispatcher : DefaultDispatcher
+    {
+        private int DefaultCount = 12;
+
+        internal AggregatedDispatcher(IDispatcher dispatcher):base(dispatcher)
+        {
+            Dispatcher = dispatcher;
+        }
+
+        internal void Flush()
+        {
+            bool first = true;
+            List<Tuple<string, string>> FlatList = new List<Tuple<string, string>>();
+            foreach (KeyValuePair<string, List<EventsBase>> pair in ObjectsToBeDispatched)
+            {
+                foreach (EventsBase Event in pair.Value)
+                {
+                    List<Tuple<string, string>> EventList = Event.GetEvents();
+                    if (first)
+                    {
+                        FlatList.AddRange(EventList);
+                        first = false;
+                        continue;
+                    }
+                    for (int i = DefaultCount; i < EventList.Count; i++)
+                    {
+                        FlatList.Add(EventList[i]);
+                    }
```

Comment:
GetRange returns a list and not list elements. In this case it would return list of tuples and not tuples.

---
In reply to: [74545633](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/494#discussion_r74545633) [](ancestors = 74545633)